### PR TITLE
test: add coverage for JSON deserialization in api_request

### LINKED ISSUE #98: Add test coverage for JSON deserialization in api_request
There is currently no test coverage for the JSON deserialization and error management in the  method of . This is critical for ensuring the robustness of data handling. It would be beneficial to add tests that cover various scenarios including successful parsing and handling of JSONDecodeError exceptions.

Referenced lines: https://github.com/uilibs/uiprotect/blob/main/src/uiprotect/api.py#L437-L442

This issue was identified during a pull request review: https://github.com/uilibs/uiprotect/pull/96#discussion_r1649735092

@bdraco

### PR DESCRIPTION
**Pull request by RaHehl**
### Description of change

Closes #98

- add tests for successful JSON parsing (dict, list, empty response)
- add tests for JSONDecodeError handling (invalid, truncated, binary)
- add tests for api_request_obj and api_request_list type validation
- add simple_api_client fixture to conftest.py for reusability
### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] This pull request follows the [contributing guidelines](https://github.com/uilibs/uiprotect/blob/main/CONTRIBUTING.md).
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/), such as "fix(api): prevent racing of requests".

> - If pre-commit.ci is failing, try `pre-commit run -a` for further information.
> - If CI / test is failing, try `poetry run pytest` for further information.

## Summary by CodeRabbit

* **Tests**
  * Added a test fixture that provides a real API client for bootstrap and cleanup in the test environment
  * Added comprehensive JSON deserialization tests for API response parsing, covering success cases, nested structures, Unicode, and error/type validation scenarios

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by RaHehl** [comment]
@coderabbitai review

File: tests/test_api.py [hunk 1]
```diff
@@ -3643,6 +3643,118 @@ def test_led_settings_serialization_with_all_fields():
     assert serialized["floodLed"] is False
 
 
+class TestApiRequestJsonDeserialization:
```
Comment (bdraco) [inline tests/test_api.py]: Ideally new tests are not classes and are bare pytest
